rsconnect/api.py

```diff
@@ -390,20 +396,20 @@ def output_task_log(task_status, last_status, log_callback):
 class RSConnectExecutor:
     def __init__(
         self,
-        ctx: click.Context = None,
-        name: str = None,
-        url: str = None,
-        api_key: str = None,
+        ctx: Optional[click.Context] = None,
```

Comment:
These instances of `Optional` were necessary to satisfy pyright's strict mode, and they make the types more accurately reflect the actual possible values.